src/toil/test/lib/dockerTest.py

```diff
@@ -37,66 +39,71 @@ def testDockerClean(self, caching=True):
         """
         Run the test container that creates a file in the work dir, and sleeps for 5 minutes.  Ensure
         that the calling job gets SIGKILLed after a minute, leaving behind the spooky/ghost/zombie
-        container. Ensure that the container is killed on batch system shutdown (through the defer
+        container. Ensure that the container is killed on batch system shutdown (through the deferParam
         mechanism).
         This inherently also tests _docker
         :returns: None
         """
-        # We need to test the behaviour of `defer` with `rm` and `detached`. We do not look at the case
+
+        # We need to test the behaviour of `deferParam` with `rm` and `detached`. We do not look at the case
         # where `rm` and `detached` are both True.  This is the truth table for the different
         # combinations at the end of the test. R = Running, X = Does not exist, E = Exists but not
         # running.
         #              None     FORGO     STOP    RM
         #    rm        X         R         X      X
         # detached     R         R         E      X
         #  Neither     R         R         E      X
-        assert os.getuid() != 0, "Cannot test this if the user is root."
```

Comment:
Should this maybe be replaced with a slightly cleverer check to see if Docker works for the current user?

Reply:
This was deleted for a reason.  Tests should not fail if the user is not root.  Instead it checks only if the end files have the correct permissions now.  Hopefully that addresses the issue raised.